pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,224 @@
+
+    ..todo:: better normalization between channels
+
+    Parameters
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    fs : float
+        Sampling rate, in Hz.
+    freqs : 1d array or list of float
+        If array, frequency values to estimate with morlet wavelets.
+        If list, define the frequency range, as [freq_start, freq_stop, freq_step].
+        The `freq_step` is optional, and defaults to 1. Range is inclusive of `freq_stop` value.
+    n_cycles : float or 1d array
+        Length of the filter, as the number of cycles for each frequency.
+        If 1d array, this defines n_cycles for each frequency.
+    scaling : float
+        Scaling factor.
+    norm : {'sss', 'amp'}, optional
+        Normalization method:
+
+        * 'sss' - divide by the square root of the sum of squares
+        * 'amp' - divide by the sum of amplitudes
+
+    Returns
+    -------
+    mwt : 2d array
+        Time frequency representation of the input signal.
+
+    Notes
+    -----
+    This computes the continuous wavelet transform at specified frequencies across time.
+    """
+    if not isinstance(sig, nap.Tsd) and not isinstance(sig, nap.TsdFrame):
+        raise TypeError("`sig` must be instance of Tsd or TsdFrame")
+    if isinstance(freqs, (tuple, list)):
+        freqs = _create_freqs(*freqs)
+    n_cycles = _check_n_cycles(n_cycles, len(freqs))
+    if isinstance(sig, nap.Tsd):
+        mwt = np.zeros([len(freqs), len(sig)], dtype=complex)
+        for ind, (freq, n_cycle) in enumerate(zip(freqs, n_cycles)):
+            mwt[ind, :] = _convolve_wavelet(sig, fs, freq, n_cycle, scaling, norm=norm)
+        return nap.TsdFrame(t=sig.index, d=np.transpose(mwt))
+    else:
+        mwt = np.zeros(
+            [sig.values.shape[0], len(freqs), sig.values.shape[1]], dtype=complex
+        )
+        for channel_i in range(sig.values.shape[1]):
+            for ind, (freq, n_cycle) in enumerate(zip(freqs, n_cycles)):
+                mwt[:, ind, channel_i] = _convolve_wavelet(
+                    sig[:, channel_i], fs, freq, n_cycle, scaling, norm=norm
+                )
+        return nap.TsdTensor(t=sig.index, d=mwt)
+
+
+def _convolve_wavelet(
+    sig, fs, freq, n_cycles=7, scaling=0.5, wavelet_len=None, norm="sss"
+):
+    """
+    Convolve a signal with a complex wavelet.
+
+    Parameters
+    ----------
+    sig : pynapple.Tsd
+        Time series to filter.
+    fs : float
+        Sampling rate, in Hz.
+    freq : float
+        Center frequency of bandpass filter.
+    n_cycles : float, optional, default: 7
+        Length of the filter, as the number of cycles of the oscillation with specified frequency.
+    scaling : float, optional, default: 0.5
+        Scaling factor for the morlet wavelet.
+    wavelet_len : int, optional
+        Length of the wavelet. If defined, this overrides the freq and n_cycles inputs.
+    norm : {'sss', 'amp'}, optional
```

Comment:
is `sss` standard?

Reply:
Both `sss` and `amp` are taken from the neurodsp package [compute_wavelet_transform](https://neurodsp-tools.github.io/neurodsp/generated/neurodsp.timefrequency.compute_wavelet_transform.html) function. They are not present in the scipy and PyWavelets compute_wavelet_transform functions, which do not have normalisation procedures built into them. 

In LFP literature, I have not seen normalisation like `sss` before on wavelets. I have seen `amp` be used, as well as something called `mad` normalisation be used. Should I remove `sss`?

Reply:
I will investigate further and add docs. These normalisation approaches both make the decomposition look a lot better, and PyWavelets (which most people who use wavelet will be familiar with) uses `sss` by default.